pinot-minion/src/main/java/com/linkedin/pinot/minion/executor/BaseTaskExecutor.java

```diff
@@ -32,4 +37,20 @@ public void setMinionContext(@Nonnull MinionContext minionContext) {
   public void cancel() {
     _cancelled = true;
   }
+
+  @Override
+  public int uploadSegment(String uri, final String fileName, final InputStream inputStream, final long lengthInBytes,
+      FileUploadUtils.SendFileMethod httpMethod, String originalSegmentCrc) {
+    List<Header> headers = new ArrayList<>();
+    Header header = new Header("If-Match", originalSegmentCrc);
+
+    // A pragma header is an implementation specific field that may have various effects along the
+    // request response chain. We will tell the controller minion is sending the segment.
+    Header minionHeader = new Header("Pragma", "Minion");
```

Comment:
Pragma is an old http 1.0 compat header, and seems to be not he right one to use here. 

I think you can use "User-Agent", and set the value to "pinot-minion-<version>". Otherwise, you can use a header like "X-Sender-Identity" (or, anything prefixed with an X). Let us chat about this and reach a conclusion.

Reply:
As Subbu mentioned, the `User-Agent` header is the correct header to use. In this case, what you want to do is to grab the default user agent value (see `DefaultHttpParams.getDefaultParams()`) and append something that indicates the minion version and/or job version, so that it looks something like this: `Jakarta Commons-HttpClient/3.1 Pinot-Minion/1.0`.

This way, if for some reason in the future we need to do some special processing for different versions of the HttpClient library or the minion code, then we can check exactly which version is used as opposed to having a single version.

Also, is the code generic for all minion based pushes or only for certain job types? If so, it might make more sense to specify the job type instead (or even both, eg. `Pinot-Minion/1.0 Pinot-Job-Foobar/1.2`).

Reply:
That's a good idea. I can add specific job types. Thanks